### PR TITLE
explorer: improve exception handling

### LINKED ISSUE #1261: on error, cannot close plugin wait dialog, need to kill the entire IDA process
on error, cannot close plugin wait dialog, need to kill the entire IDA process

_Originally posted by @mr-tz in https://github.com/mandiant/capa/issues/1254#issuecomment-1369097019_

### PR DESCRIPTION
**Pull request by mike-hunhoff**
Closes #1261

Improved exception handling to prevent IDA from locking up if certain errors occur during execution. User is presented with a dialog indicating an error occurred and they should see the output window for more details.

Screen Shot 2023-01-04 at 10 28 21 AM

### CODE REVIEW

File: capa/ida/plugin/form.py [hunk 1]
```diff
@@ -673,19 +683,20 @@ def slot_progress_feature_extraction(text):
...
             if ida_kernwin.user_cancelled():
                 logger.info("User cancelled analysis.")
                 return False
 
             update_wait_box("extracting features")
 
             try:
+                # just generated above
+                assert self.ruleset_cache is not None
+
```
Comment (mike-hunhoff) [inline capa/ida/plugin/form.py:699]: moved `assert` statements under `try`/`except` blocks so we can exit gracefully and not lock up IDA

File: capa/ida/plugin/form.py [hunk 2]
```diff
@@ -656,15 +656,25 @@ def slot_progress_feature_extraction(text):
...
+                logger.info("User cancelled analysis.")
+                return False
 
             update_wait_box("calculating analysis")
 
             try:
                 self.process_total += len(tuple(extractor.get_functions()))
             except Exception as e:
-                logger.error("Failed to calculate analysis (error: %s).", e)
+                logger.error("Failed to calculate analysis (error: %s).", e, exc_info=True)
```
Comment (mr-tz) [inline capa/ida/plugin/form.py:677]: I like these additions

File: capa/ida/plugin/form.py [hunk 3]
```diff
@@ -815,13 +821,13 @@ def load_capa_function_results(self):
             logger.info("User cancelled analysis.")
             return False
 
-        update_wait_box("loading IDA extractor")
+        update_wait_box("initializing feature extractor")
```
Comment (mr-tz) [inline capa/ida/plugin/form.py:824]:
```suggestion
        update_wait_box("Initializing feature extractor")
```

File: capa/ida/plugin/form.py [hunk 4]
```diff
@@ -815,13 +821,13 @@ def load_capa_function_results(self):
...
 
         try:
             # must use extractor to get function, as capa analysis requires casted object
             extractor = CapaExplorerFeatureExtractor()
         except Exception as e:
-            logger.error("Failed to load IDA feature extractor (error: %s)", e)
+            logger.error("Failed to init feature extractor (error: %s)", e, exc_info=True)
```
Comment (mr-tz) [inline capa/ida/plugin/form.py:830]:
```suggestion
            logger.error("Failed to initialize feature extractor (error: %s)", e, exc_info=True)
```

**Review by mr-tz — APPROVED** [review]
Only spelling suggestions. Thank you. Super annoying when IDA freezes.

**Pull request merged**